pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -26,55 +29,74 @@ import (
 	osclient "github.com/openshift/origin/pkg/client"
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	"github.com/openshift/origin/pkg/generate/git"
+	"github.com/openshift/source-to-image/pkg/tar"
 )
 
 const (
 	startBuildLong = `
 Start a build
 
-This command starts a build for the provided BuildConfig or re-runs an existing build using
---from-build=<name>. You may pass the --follow flag to see output from the build.`
+This command starts a new build for the provided build config or copies an existing build using
+--from-build=<name>. Pass the --follow flag to see output from the build.
 
-	startBuildExample = `  # Starts build from BuildConfig matching the name "3bd2ug53b"
-  $ %[1]s start-build sample-build
+In addition, you can pass a file, directory, or source code repository with the --from-file,
+--from-dir, or --from-repo flags directly to the build. The contents will be streamed to the build
+and override the current build source settings. When using --from-repo, the --commit flag can be
+used to control which branch, tag, or commit is sent to the server. If you pass --from-file, the
+file is placed in the root of an empty directory with the same filename. Note that builds
+triggered from binary input will not preserve the source on the server, so a rebuild is not
+automatic.
+`
 
-  # Starts build from build matching the name "sample-build-1"
-  $ %[1]s start-build --from-build=sample-build-1
+	startBuildExample = `  # Starts build from build config "hello-world"
+  $ %[1]s start-build hello-world
 
-  # Starts build from BuildConfig matching the name "sample-build" and watches
-	# the logs until the build completes or fails
-  $ %[1]s start-build sample-build --follow
+  # Starts build from a previous build "hello-world-1"
+  $ %[1]s start-build --from-build=hello-world-1
 
-  # Starts build from BuildConfig matching the name "sample-build" and wait until
-	# the build completes. It exits with a non-zero return code if the build
-	# fails. 
-  $ %[1]s start-build sample-build --wait`
+  # Use the contents of a directory as build input
+  $ %[1]s start-build hello-world --from-dir=src/
+
+  # Send the contents of a Git repository to the server from tag 'v2'
+  $ %[1]s start-build hello-world --from-repo=../hello-world --commit=v2
```

Comment:
--from-local-repo would be more accurate, though.  At a minimum the help text should be more explicit.